src/test/java/com/jcabi/log/VerboseProcessTest.java

```diff
@@ -477,4 +481,125 @@ public int decide(final LoggingEvent event) {
+        MatcherAssert.assertThat(thrownException.getMessage(),
+                Matchers.containsString(VerboseProcessExample.SYSOUT_1));
+        MatcherAssert.assertThat(thrownException.getMessage(),
+                Matchers.containsString(VerboseProcessExample.SYSOUT_2));
+        MatcherAssert.assertThat(thrownException.getMessage(),        
+        Matchers.containsString(VerboseProcessExample.SYSOUT_3));
+        MatcherAssert.assertThat(thrownException.getMessage(),
+                Matchers.containsString(VerboseProcessExample.SYSOUT_4));
+        boolean foundCompleteStack = false;
+        for (final LoggingEvent event : appender.getLog()) {
+            final String message = (String) event.getMessage();
+            if (message.contains(VerboseProcessExample.THROWN_ERR_MSG)) {
+                final boolean containsCaughtException =
+                        message.contains(VerboseProcessExample.CAUGHT_ERR_MSG);
+                Assert.assertTrue(containsCaughtException);
+                foundCompleteStack = containsCaughtException;
+            }
+        }
+        Assert.assertTrue(foundCompleteStack);
+    }
+
+    /**
+     * Gets the location of Java, whether on Linux of Windows.
+     * @return
+     */
+    public static String getJavaExecLoc() {
+        String javaExec = null;
+        String sunLibPath = System.getProperty("java.home");
+        javaExec = new String();
+        if (SystemUtils.IS_OS_WINDOWS) {
+            // Windows
+            try {
+                final String sunLibPathNew =
+                        sunLibPath.replaceAll("\\\\", "\\\\\\\\");
+                sunLibPath = sunLibPathNew;
+                javaExec = (sunLibPath + "\\bin\\java.exe");
+                // Test Location
+                final File file1 = new File(javaExec);
+                if (file1.exists()) {
+                    System.out.println("File [" + javaExec + "] exists.");
+                    return javaExec;
+                }
+            } catch (final Exception e) {
+                throw new RuntimeException("Unable to get the Java Path", e);
+            }
+        } else {
+            // Unix
+            try {
+                javaExec = sunLibPath + "/bin/java";
+                // Test Location
+                final File file1 = new File(javaExec);
+                if (file1.exists()) {
+                    return javaExec;
+                }
+            } catch (final Exception e) {
+                throw new RuntimeException("Unable to get the Java Path", e);
+            }
+        }
+        throw new RuntimeException("Unable to get the Java Path.");
+    }
+
+    /**
+     * Logger appender that compiles a list of all LoggingEvents.
+     * @author dean.e.clark
+     */
+    class TestAppender extends AppenderSkeleton {
+        /**
+         * List of logging events.
+         */
+        private final List<LoggingEvent> log = new ArrayList<LoggingEvent>();
```

Comment:
@dean-e-clark we need to come up with an initial size here to satisfy the style, but I know it doesn't matter here that much